log.go

```diff
@@ -80,6 +96,107 @@ type eventLogger struct {
 	// TODO add log-level
 }
 
+type activeSpanKeyType struct{}
+
+var activeSpanKey = activeSpanKeyType{}
+
+type activeSpan struct {
+	isFinished bool
+}
+
+func (el *eventLogger) Start(ctx context.Context, name string) context.Context {
+	as := &activeSpan{
+		isFinished: false,
+	}
+	ctx = context.WithValue(ctx, activeSpanKey, as)
+	span, ctx := opentrace.StartSpanFromContext(ctx, name)
+	span.SetTag("system", el.system)
+	return ctx
+}
+
+func (el *eventLogger) StartFromParentState(ctx context.Context, name string, parent []byte) (context.Context, error) {
+	sc, err := deserializeContext(parent)
+	if err != nil {
+		return nil, err
+	}
+
+	as := &activeSpan{
+		isFinished: false,
+	}
+	ctx = context.WithValue(ctx, activeSpanKey, as)
+	span, ctx := opentrace.StartSpanFromContext(ctx, name, otExt.RPCServerOption(sc))
+	span.SetTag("system", el.system)
+	return ctx, nil
+}
+
+func (el *eventLogger) SerializeContext(ctx context.Context) ([]byte, error) {
+	gTracer := opentrace.GlobalTracer()
+	var carrier bytes.Buffer
+	span := opentrace.SpanFromContext(ctx)
+	if err := gTracer.Inject(span.Context(), opentrace.Binary, carrier); err != nil {
+		return nil, errors.New("Failed to serialize context")
+	}
+	return carrier.Bytes(), nil
+}
+
+func (el *eventLogger) LogKV(ctx context.Context, key string, value interface{}) {
+	span := opentrace.SpanFromContext(ctx)
+	if span != nil {
```

Comment:
I'd still log *something* in the else case. Either with `el.Errorf` or by creating a fake span (may be easier to track?).